src/time_integration/paired_explicit_runge_kutta/methods_PERK2.jl

```diff
@@ -0,0 +1,361 @@
+    a_matrix[:, 1] = c[3:end]
+
+    cons_order = 2
+    filter_thres = 1e-12
+    dtmax = 1.0
+    dt_eps = 1e-9
+
+    eig_vals = eigvals(jacobian_ad_forward(semi))
+
+    num_eig_vals, eig_vals = filter_eigvals(eig_vals, filter_thres)
+
+    mon_coeffs, dt_opt = bisection(cons_order, num_eig_vals, num_stages, dtmax, dt_eps,
+                                   eig_vals)
+    mon_coeffs = undo_normalization!(cons_order, num_stages, mon_coeffs)
+
+    num_mon_coeffs = length(mon_coeffs)
+    @assert num_mon_coeffs == coeffs_max
+    A = compute_a_coeffs(num_stages, se_factors, mon_coeffs)
+
+    a_matrix[:, 1] -= A
+    a_matrix[:, 2] = A
+
+    println("A matrix: ")
+    display(a_matrix)
+    println()
+
+    return a_matrix, c
+end
+
+function compute_PERK2_Butcher_tableau(num_stages, base_path_mon_coeffs::AbstractString,
+                                       bS, c_end)
+
+    # c Vector form Butcher Tableau (defines timestep per stage)
+    c = zeros(num_stages)
+    for k in 2:num_stages
+        c[k] = c_end * (k - 1) / (num_stages - 1)
+    end
+    println("Timestep-split: ")
+    display(c)
+    println("\n")
+    se_factors = bS * reverse(c[2:(end - 1)])
+
+    # - 2 Since First entry of A is always zero (explicit method) and second is given by c_2 (consistency)
+    coeffs_max = num_stages - 2
+
+    a_matrix = zeros(coeffs_max, 2)
+    a_matrix[:, 1] = c[3:end]
+
+    path_mon_coeffs = base_path_mon_coeffs * "gamma_" * string(num_stages) * ".txt"
+    num_mon_coeffs, mon_coeffs = read_file(path_mon_coeffs, Float64)
+    @assert num_mon_coeffs == coeffs_max
+    A = compute_a_coeffs(num_stages, se_factors, mon_coeffs)
+
+    a_matrix[:, 1] -= A
+    a_matrix[:, 2] = A
+
+    println("A matrix: ")
+    display(a_matrix)
+    println()
+
+    return a_matrix, c
+end
+
+"""
+    PERK2()
+
+The following structures and methods provide a minimal implementation of
+the second-order paired explicit Runge-Kutta method
+optimized for a certain simulation setup (PDE, IC & BC, Riemann Solver, DG Solver).
+
+- Vermeire (2019).
+Paired explicit Runge-Kutta schemes for stiff systems of equations
+[DOI: 10.1016/j.jcp.2019.05.014](https://doi.org/10.1016/j.jcp.2019.05.014)
+"""
+
+mutable struct PERK2 <: PERKSingle
+    const num_stages::Int
+
+    a_matrix::Matrix{Float64}
+    c::Vector{Float64}
+    bS::Float64
+    b1::Float64
+    c_end::Float64
+
+    #Constructor that read the coefficients from the file
+    function PERK2(num_stages, base_path_mon_coeffs::AbstractString, bS = 1.0,
+                   c_end = 0.5)
+        newPERK2 = new(num_stages)
+
+        newPERK2.a_matrix, newPERK2.c = compute_PERK2_Butcher_tableau(num_stages,
+                                                                      base_path_mon_coeffs,
+                                                                      bS, c_end)
+
+        newPERK2.b1 = one(bS) - bS
+        newPERK2.bS = bS
+        newPERK2.c_end = c_end
+        return newPERK2
+    end
+
```

Comment:
Can you add an additional constructor which takes in a list of eigenvalues and then calls a version with the optimizer? Also, we should refactor

`function compute_PERK2_Butcher_tableau(num_stages, semi::AbstractSemidiscretization,` 

by extracting the optimization given the eigenvalues.

Reply:
I have done this in the latest commit. Please let me know if you want me to edit anything!